proto/minder/v1/minder.proto

```diff
@@ -2163,45 +2284,97 @@ message RuleType {
+                        max_len: 1000,
+                    },
+                    (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+                ];
             }
 
             // the name stutters a bit but we already use a PullRequest message for handling PR entities
             message PullRequestRemediation {
                 message Content {
                     // the file to patch
-                    string path = 1;
+                    string path = 1 [
+                        (buf.validate.field).string = {
+                            pattern: "^\\.?([[:word:].-]+\\/)*[[:word:].-]+(?:\\.[[:alnum:]]+)?$",
+                            min_len: 1,
+                            max_len: 200,
+                        }
+                    ];
                     // how to patch the file. For now, only replace is supported
-                    string action = 2;
+                    string action = 2 [
+                        (buf.validate.field).string = {
+                            in: ["replace"],
+                            min_len: 1,
+                            max_len: 50,
+                        }
+                    ];
                     // the content of the file
                     string content = 4;
                     // the GIT mode of the file. Not UNIX mode! String because the GH API also uses strings
                     // the usual modes are: 100644 for regular files, 100755 for executable files and
                     // 040000 for submodules (which we don't use but now you know the meaning of the 1 in 100644)
                     // see e.g. https://github.com/go-git/go-git/blob/32e0172851c35ae2fac495069c923330040903d2/plumbing/filemode/filemode.go#L16
-                    optional string mode = 3;
+                    optional string mode = 3 [
+                        (buf.validate.field).string = {
+                            pattern: "^\\d+$",
+                            max_len: 6,
+                        },
+                        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+                    ];
                 }
 
                 message ActionsReplaceTagsWithSha {
                     // List of actions to exclude from the replacement
-                    repeated string exclude = 1;
+                    repeated string exclude = 1 [
+                        (buf.validate.field).repeated = {
+                            items: {
+                                string: {
+                                    pattern: "^\\.?([[:word:].-]+\\/)*[[:word:].-]+(?:\\.[[:alnum:]]+)?$",
+                                    max_len: 200,
+                                }
+                            }
+                        }
+                    ];
                 }
 
                 // the title of the PR
-                string title = 1;
+                string title = 1 [
+                    (buf.validate.field).string = {
```

Comment:
No pattern here?  That could be okay, given that GitHub will validate this, but you should put a comment as to why no pattern.